Assets/MixedRealityToolkit.SDK/Experimental/Features/Support/Scripts/BaseServiceManager.cs

```diff
@@ -0,0 +1,395 @@
+                }
+
+                IReadOnlyList<IMixedRealityService> serviceSnapshot = new List<IMixedRealityService>(registeredServices.Values);
+                for (int i = 0; i < serviceSnapshot.Count; i++)
+                {
+                    serviceSnapshot[i]?.Disable();
+                }
+            }
+        }
+
+        protected virtual void OnDestroy()
+        {
+            IMixedRealityDataProvider[] providers = dataProviders.ToArray();
+            for (int i = 0; i < providers.Length; i++)
+            {
+                providers[i]?.Disable(); // Disable before destroy to ensure the data provider has time to get in a good state.
+                providers[i]?.Destroy();
+            }
+            // Clear the actual collection
+            dataProviders.Clear();
+
+            IReadOnlyList<IMixedRealityService> serviceSnapshot = new List<IMixedRealityService>(registeredServices.Values);
+            for (int i = 0; i < serviceSnapshot.Count; i++)
+            {
+                serviceSnapshot[i].Disable(); // Disable before destroy to ensure the service has time to get in a good state.
+                serviceSnapshot[i].Destroy();
+            }
+            // Clear the actual collection
+            registeredServices.Clear();
+        }
+
+        #endregion MonoBehaviour implementation
+
+        #region IMixedRealityServiceRegistrar implementation
+
+        /// <inheritdoc />
+        public T GetDataProvider<T>(string name = null) where T : IMixedRealityDataProvider
+        {
+            Type interfaceType = typeof(T);
+            T provider = default(T);
+
+            IMixedRealityDataProvider[] providers = dataProviders.ToArray();
+            for (int i = 0; i < providers.Length; i++)
+            {
+                // Check for null and mismatched type.
+                if ((providers[i] == null) || !interfaceType.IsAssignableFrom(providers[i].GetType())) { continue; }
+
+                // Check to see if name is valid (not null or whitespace) and if it matches the provider's name.
+                if (!string.IsNullOrWhiteSpace(name) && string.Equals(providers[i].Name, name))
+                {
+                    provider = (T)providers[i];
+                }
+                // If name is invalid, the first instance of a matching provider type indicates that a registration has occurred.
+                else
+                {
+                    provider = (T)providers[i];
+                }
+            }
+
+            return provider;
+        }
+
+        /// <inheritdoc />
+        public IReadOnlyList<T> GetDataProviders<T>(string name = null) where T : IMixedRealityDataProvider
+        {
+            if (!string.IsNullOrWhiteSpace(name))
+            {
+                Debug.LogError("This registrar does not support requesting multiple data providers of the same interface type and name.");
+                return new List<T>(); ;
+            }
```

Comment:
Why is this one different from the defined functionality of the base [https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/mrtk_development/Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs)?

It seems like if we inheritdoc and implement the interface, we should honor the contract. Otherwise, we shouldn't implement the interface. Or, provide more guidance about why this one in particular doesn't adhere to the existing interface.

Reply:
The interface was designed to support the MRTK object's public methods. we should consider removing this one as it is of limited use.

Reply:
while retaining the MRTK object's behavior

Reply:
removed the short circuit and added a full implementation